### PR TITLE
SOLR-14429: Convert XXX.txt files to proper XXX.md

### PR DESCRIPTION
**Pull request by mocobeta**
# Description

Converted all README.txt to README.md and partially fixed its formatting (as proper markdown). I also fixed pointers to the files.

See https://issues.apache.org/jira/browse/SOLR-14429

# Tests

- Distribution package built by `ant create-package` includes the all .md files
- ~TODO: run smoke test~ This branch passed `nightly-smoke`
```
[smoker] SUCCESS! [0:43:25.703442]
```

### CODE REVIEW

**Comment by madrob** [comment]
Do we need to update anything in `rat-sources.gradle` or `check-source-paterns.groovy`?

File: solr/example/README.md [hunk 1]
```diff
@@ -1,57 +1,74 @@
...
 
 For more information about this example please read...
 
- * example/solr/README.txt
-   For more information about the "Solr Home" and Solr specific configuration
- * https://lucene.apache.org/solr/guide/solr-tutorial.html
-   For a Solr tutorial
- * http://wiki.apache.org/solr/SolrResources 
-   For a list of other tutorials and introductory articles.
+ * example/solr/README.md
```
Comment (dsmiley) [inline solr/example/README.md:61]: That path looks wrong; probably meant solr/example/README.md and it's a reference to the current document.  Can you show as a link easily?  If not; no big deal.

File: solr/example/files/README.md [hunk 2]
```diff
@@ -111,38 +120,46 @@ For further explanations, see the frequently asked questions at the end of the g
...
-        under server/solr/files/conf/velocity.
+    ```
+    bin/solr start -Dvelocity.template.base.dir=</full/path/to>/example/files/conf/velocity/
+    ```
+ 
+If you want to adjust the browse templates for an existing collection, edit the core’s configuration
+under server/solr/files/conf/velocity.
 
 
 =======
```
Comment (dsmiley) [inline solr/example/files/README.md:165]: At least on GitHub, this isn't showing as markup.  (I assume this line is at the end directly about the provenance info)

File: solr/example/README.md (hunk 1 above)
Comment (dsmiley) [inline solr/example/README.md:61]: This is a reference to itself and it's wrong; should be solr/example/README.md

 Not sure if you can easily do as a link but it's probably moot.

**Comment by mocobeta** [comment]
> Do we need to update anything in rat-sources.gradle or check-source-paterns.groovy?

@madrob thank you, I think so... I fixed both files (added name pattern `*.md` next to `*.txt`): https://github.com/apache/lucene-solr/pull/1450/commits/0b38051d8156d218b57b6db0581ccf0ca84fbd88; but I'm not fully sure of that. Does the change look correct?

File: solr/example/README.md (hunk 1 above)
Comment (mocobeta) [reply solr/example/README.md]: I fixed it to a link to the file itself; though I'm not sure there is meaning in this link. It looks like this: https://github.com/mocobeta/lucene-solr/blob/solr-change-txt-to-md/solr/example/README.md
We are probably able to just remove this line...?

File: solr/example/files/README.md (hunk 2 above)
Comment (mocobeta) [reply solr/example/files/README.md]: Thanks, I missed it. It was fixed as this: https://github.com/mocobeta/lucene-solr/tree/solr-change-txt-to-md/solr/example/files

**Comment by mocobeta** [comment]
@dsmiley 
> solr/contrib/analysis-extras/README.md could use some markup to reflect the items as a bulleted list.

I tried to reformat the file, does this look better than before?
https://github.com/mocobeta/lucene-solr/blob/solr-change-txt-to-md/solr/contrib/analysis-extras/README.md

(Maybe we should have a good README template for `contrib` modules :slightly_smiling_face:

**Comment by mocobeta** [comment]
Hi @dsmiley, 
I just wanted to inform you that I updated the branch. Any comments, or can I merge it to the master?

**Comment by mocobeta** [comment]
Merged, thank you for reviewing.

**Pull request merged**